File: jvm/src/main/kotlin/com/jetbrains/signatureverifier/serialization/dataholders/attributeInfos/CertificationAuthorityAttributeInfo.kt
package com.jetbrains.signatureverifier.serialization.dataholders

import com.jetbrains.signatureverifier.serialization.toPrimitiveDLSet
import kotlinx.serialization.Serializable
import org.bouncycastle.asn1.cms.Attribute
import org.bouncycastle.asn1.x509.AlgorithmIdentifier

@Serializable
data class CertificationAuthorityAttributeInfo(
  override val identifier: TextualInfo,
  val content: List<AlgorithmInfo>
) : AttributeInfo() {

  constructor(attribute: Attribute) : this(
    TextualInfo.getInstance(attribute.attrType),
    attribute.attributeValues.map { AlgorithmInfo(AlgorithmIdentifier.getInstance(it)) }
  )

  override fun getPrimitiveContent() = content.toPrimitiveDLSet()
}

File: net/JetBrains.Serialization/src/Dataholders/CertificationAuthorityAttributeInfo.cs
using Newtonsoft.Json;
using Org.BouncyCastle.Asn1;
using Attribute = Org.BouncyCastle.Asn1.Cms.Attribute;
using JetBrains.Serialization;
using Org.BouncyCastle.Asn1.X509;

[JsonObject(MemberSerialization.Fields)]
public class CertificationAuthorityAttributeInfo : AttributeInfo
{
  public override TextualInfo Identifier { get; }
  public List<AlgorithmInfo> Content { get; }

  public CertificationAuthorityAttributeInfo(Attribute attribute)
  {
    Identifier = TextualInfo.GetInstance(attribute.AttrType);
    Content = attribute.AttrValues.ToArray().Select(item => new AlgorithmInfo(AlgorithmIdentifier.GetInstance(item)))
      .ToList();
  }

  public override Asn1Encodable GetPrimitiveContent() => Content.ToPrimitiveDerSet();
}